cookbook/case_studies/house_price_prediction/house_price_predictor.py

```diff
@@ -0,0 +1,253 @@
+"""
+Predicting House Price in a Region Using an XGBoost Model and Flytekit (Python)
+-------------------------------------------------------------------------------
+
+"""
+
+# %%
+# Install the following three libraries before running the model (locally):
+# .. code-block:: python
+#
+#       pip install sklearn
+#       pip install joblib
+#       pip install xgboost
+
+# %%
+# Step 1: Importing the Libraries
+# -------------------------------
+import os
+import typing
+
+import joblib
+import numpy as np
+import pandas as pd
+from sklearn.metrics import accuracy_score, mean_absolute_error
+from sklearn.model_selection import train_test_split
+from xgboost import XGBClassifier
+from flytekit import Resources, dynamic, task, workflow
+from flytekit.types.file import FlyteFile
+from flytekit.types.directory import FlyteDirectory
+
+# %%
+# Step 2: Initializing the Variables
+# ----------------------------------
+NUM_HOUSES_PER_LOCATION = 1000
+LOCATIONS = [
+    "NewYork_NY",
+    "LosAngeles_CA",
+    "Chicago_IL",
+    "Houston_TX",
+    "Dallas_TX",
+    "Phoenix_AZ",
+    "Philadelphia_PA",
+    "SanAntonio_TX",
+    "SanDiego_CA",
+    "SanFrancisco_CA",
+]
+
+MAX_YEAR = 2021
+SPLIT_RATIOS = [0.6, 0.3, 0.1]
+
+# %%
+# Step 3: Defining the Data Generation Functions
+# ----------------------------------------------
+#
+# House price generation function:
+def gen_price(house) -> int:
+    _base_price = int(house["SQUARE_FEET"] * 150)
+    _price = int(
+        _base_price
+        + (10000 * house["NUM_BEDROOMS"])
+        + (15000 * house["NUM_BATHROOMS"])
+        + (15000 * house["LOT_ACRES"])
+        + (15000 * house["GARAGE_SPACES"])
+        - (5000 * (MAX_YEAR - house["YEAR_BUILT"]))
+    )
+    return _price
+# %%
+# Random data pertaining to a house:
+def gen_random_house() -> typing.List:
```

Comment:
how about converting this to a pandas dataframe?

Reply:
cc @cosmicBboy

Reply:
basically we can collapse the 2 functions into one

Reply:
`gen_random_house()` function is to generate the house details for one house, and the `gen_houses` function is to generate all the house details which internally calls the former function.

Is there a necessity to merge them?

Reply:
Done.